src/features/editors/VisualHTMLEditor.tsx

```diff
@@ -0,0 +1,92 @@
+import './VisualHTMLEditor.css'
+import {CodeSegment, CodeSegments} from "../../editorconfig"
+import Line from "./VisualHTMLEditorLine"
+import {DragDropContext, DropResult} from 'react-beautiful-dnd'
+import {useAppDispatch} from "../../hooks"
+import {moveElement} from "../../store/features/editors/visualHTML";
+
+interface IVisualHTMLEditor {
+  elements: CodeSegments
+}
+
+export type LineSegment = {
+  id: string,
+  type: string,
+  value: string,
+  unlocked: boolean,
+  index: number
+}
+
+export type LineSegments = Array<LineSegment>
+
+type TLine = {
+  lineContents: LineSegments,
+  lineIndentation: number
+}
+
+function VisualHTMLEditor(props: IVisualHTMLEditor) {
+  let indentCounter = 0
+  let lineBuilder: LineSegments = []
+  let lines: Array<TLine> = []
+  let prevWasBlock: boolean = true
+
+  const dispatch = useAppDispatch()
+
+  const indenter = (element: CodeSegment) => element.type === 'opening' ? indentCounter++ : indentCounter--
+
+  const onDragEnd = (result: DropResult) => {
+    if (result.source && result.destination) {
+      if (result.source.index !== result.destination.index) {
+        dispatch(moveElement(result))
+      }
+    }
+  }
+
+  function identify(tag: string) { // Identify block type to determine linebreaks
```

Comment:
The name of this function doesn't explain what it does: identify what? :smile:

Reply:
🙈 Oops! Thanks for noticing that, fixed ( --> identifyBlockType() )